api/datasets.go

```diff
@@ -247,7 +247,11 @@ func (h *DatasetHandlers) zipDatasetHandler(w http.ResponseWriter, r *http.Reque
 
 	w.Header().Set("Content-Type", "application/zip")
 	w.Header().Set("Content-Disposition", fmt.Sprintf("filename=\"%s.zip\"", "dataset"))
-	dsutil.WriteZipArchive(h.repo.Store(), ds, res.String(), w)
+	err = dsutil.WriteZipArchive(h.repo.Store(), ds, "json", res.String(), w)
```

Comment:
remove the `"json"` hardcode in favour of a `r.FormValue("format")`, also, I think the default should be `"yaml"` considering we use it exclusively in our docs ATM. We should start getting into the habit of keeping parity across our API and CLI for all changes moving forward to avoid tech debt.

Reply:
The previous behavior was using "json", because WriteZipArchive was only working that way. I'm fine with changing it, but it'll include updating the api response data.